scripts/cmake/rustlang.cmake

```
@@ -85,6 +85,20 @@ function(build_rust_archives)
         list(APPEND RUST_BUILD_CARGO_ENV MACOSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET})
     endif()
 
+    if(ANDROID)
+        get_filename_component(ANDROID_TOOLCHAIN_ROOT_BIN ${CMAKE_C_COMPILER} DIRECTORY)
+        
+        # Rust and Clang disagree about armv7, otherwise we can take the Rust build arch. 
+        if(${RUST_BUILD_ARCH} STREQUAL armv7-linux-androideabi)
+            set(ANDROID_ARCH_NAME armv7a-linux-androideabi)
+        else()
+            set(ANDROID_ARCH_NAME ${RUST_BUILD_ARCH})
+        endif()
+
+        
+        list(APPEND RUST_BUILD_CARGO_ENV CC=${ANDROID_TOOLCHAIN_ROOT_BIN}/${ANDROID_ARCH_NAME}${ANDROID_NATIVE_API_LEVEL}-clang)
```

Wait! We need to change this to

```suggestion
        list(APPEND RUST_BUILD_CARGO_ENV RUSTFLAGS=-Clinker=${ANDROID_TOOLCHAIN_ROOT_BIN}/${ANDROID_ARCH_NAME}${ANDROID_NATIVE_API_LEVEL}-clang)
```

---

Wait! I don't think they are exclusive. 
I think we should set both, so that if a rust crate calls out to a C compiler, it uses the right one.